gramex/ml.py

```diff
@@ -377,5 +379,56 @@ def translater(handler, source='en', target='nl', key=None, cache=None, api='goo
     raise Return(result.to_json(orient='records'))
 
 
+_languagetool = {
+    'defaults': {k: v for k, v in variables.items() if k.startswith('LT_')},
+    'installed': os.path.isdir(variables['LT_CWD'])
+}
+
+
+@coroutine
+def languagetool(handler, *args, **kwargs):
+    merge(kwargs, _languagetool['defaults'], mode='setdefault')
+    # If languagetool is not installed, install it
+    if not _languagetool['installed']:
+        if not os.path.isdir(kwargs['LT_TARGET']):
+            os.makedirs(kwargs['LT_TARGET'])
+        import requests, zipfile, io        # noqa
+        src = kwargs['LT_SRC'].format(**kwargs)
+        app_log.info('Downloading languagetools from %s', src)
+        stream = io.BytesIO(requests.get(src).content)
+        app_log.info('Unzipping languagetools to %s', kwargs['LT_TARGET'])
+        zipfile.ZipFile(stream).extractall(kwargs['LT_TARGET'])
+        _languagetool['installed'] = True
+
+    # Process request via languagetool
+    client = AsyncHTTPClient()
+    tries = int(handler.get_argument('tries', 3))
+    url = kwargs['LT_URL'].format(
+        port=kwargs['LT_PORT'], language=handler.get_argument('language', 'en-us'),
+        q=six.moves.urllib_parse.quote(handler.get_argument('q', ''))
+    )
+    while tries:
+        try:
+            result = yield client.fetch(url)
+            tries = 0
+        except ConnectionRefusedError:
+            # Start languagetool
+            from gramex.cache import daemon
+            cmd = [p.format(**kwargs) for p in kwargs['LT_CMD']]
+            app_log.info('Starting: %s', ' '.join(cmd))
+            if 'proc' not in _languagetool:
+                _languagetool['proc'] = daemon(cmd, cwd=kwargs['LT_CWD'], first_line=lambda: True,
+                                               stream=True)
+                import time
+                time.sleep(5)
```

Comment:
Please explain why we are doing this instead of checking for `first_line` in a comment.